ros2bag_tools/ros2bag_tools/exporter/image.py

```diff
@@ -33,6 +34,97 @@ def conversion_code(bayer_encoding, desired_encoding, demosaicing):
         colorspace = 'BGR'
     return getattr(cv, f'COLOR_BAYER_{pattern}2{colorspace}{algorithm}')
 
+class ImageMsgWriter:
+    def __init__(self):
+        self.image_bridge = CvBridge()
+    @staticmethod
+    def get_message_type():
+        """Get the supported message type.
+        """
+        return Image
+    @staticmethod
+    def convert_encoding(img, img_encoding, encoding, demosaicing):
+        """Convert colorspace encoding and debayer if necessary.
+        """
+        if img_encoding.startswith('bayer_') and demosaicing:
+            # use custom debayering
+            # cvtColor2 unfortunately doesn't expose a demosaicing algorithm parameter
+            conv_code = conversion_code(
+                img_encoding, encoding, demosaicing)
+            img = cv.cvtColor(img, conv_code)
+        else:
+            try:
+                img = cvtColor2(img, img_encoding, encoding)
+            except RuntimeError as e:
+                raise CvBridgeError(e)
+        return img
+
+    def process(self, img_msg, encoding, demosaicing, ext, input_encoding=None, params=None):
+        """Write encoded image to buffer and return if successfull True, buf otherwise False, buf.
+        """
+        img = self.image_bridge.imgmsg_to_cv2(msg)
+        if encoding != 'passthrough':
+            in_encoding = img_msg.encoding if not input_encoding else input_encoding
+            img = ImageMsgWriter.convert_encoding(img, in_encoding, encoding, demosaicing)
+        reval, buf = cv.imencode(ext, img, params)
+        return buf
```

Comment:
Returning bool to indicate success/failure can hide details on errors from the caller. Please return the image buffer on success, and raise an error otherwise. Maybe imencode can return more information about a failure?

Reply:
Sounds good